idp/usso: allow fixing username

mhilton opened the pull request:
Add a setting that, if enabled, causes the usso identity provider to
provide stable usernames. The username will be fixed when the user first
authenticates to the system and will not be updated.

jujugui commented:
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/candid/44/

mhilton committed:
idp/usso: allow fixing username

Add a setting that, if enabled, causes the usso identity provider to
provide stable usernames. The username will be fixed when the user first
authenticates to the system and will not be updated.

mhilton committed:
Merge branch 'master' into 189-fix-openid-registration

alesstimec commented on idp/usso/usso.go:
```diff
@@ -180,55 +186,103 @@ func (idp *identityProvider) callback(ctx context.Context, w http.ResponseWriter
+		if errgo.Cause(err) == store.ErrDuplicateUsername {
+			// TODO(mhilton) format an error with useful instructions.
+			logger.Errorf("Error creating identity for %s, duplicate username %s", identity.ProviderID, username)
+			err = errgo.WithCausef(nil, params.ErrorCode("bad username"), "cannot create identity with username %s", username)
```
"bad username" seems weird.. how about "duplicate username"

mhilton replied:
Except I also want to use it in places where the username is bad in other ways. Mainly not being a valid juju username.
